cpp/dolfinx/la/Vector.h

```diff
@@ -40,15 +40,28 @@ class Vector
   }
 
   /// Copy constructor
-  Vector(const Vector& x) = default;
+  Vector(const Vector& x)
+      : _map(x._map), _bs(x._bs), _request(MPI_REQUEST_NULL),
+        _buffer_send_fwd(x._buffer_send_fwd),
+        _buffer_recv_fwd(x._buffer_recv_fwd), _x(x._x)
+  {
+    MPI_Type_dup(x._datatype, &_datatype);
+  };
 
   /// Move constructor
-  Vector(Vector&& x) noexcept = default;
+  Vector(Vector&& x)
+      : _map(std::move(x._map)), _bs(std::move(x._bs)),
+        _request(std::move(x._request)),
+        _buffer_send_fwd(std::move(x._buffer_send_fwd)),
+        _buffer_recv_fwd(std::move(x._buffer_recv_fwd)), _x(std::move(x._x))
+  {
+    MPI_Type_dup(x._datatype, &_datatype);
```

Comment:
I don't think this should duplicate. Should be similar to https://github.com/FEniCS/dolfinx/blob/43847f184f3aa69f166de5fafe9584de914b4e11/cpp/dolfinx/la/PETScVector.cpp#L202.